src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputData) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : inputData) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            String operationCode = parts[0];
+            String fruit = parts[1];
+            int quantity = Integer.parseInt(parts[2]);
+            FruitTransaction.Operation operation = getOperationFromCode(operationCode);
+            transactions.add(new FruitTransaction(operation, fruit, quantity));
+        }
+        return transactions;
+    }
+
+    private FruitTransaction.Operation getOperationFromCode(String code) {
+        for (FruitTransaction.Operation operation : FruitTransaction.Operation.values()) {
+            if (operation.getCode().equals(code)) {
+                return operation;
+            }
+        }
+        throw new IllegalArgumentException("Invalid operation code: " + code);
+    }
```

Comment:
Instead of looping through all the operations every time you want to convert a code to an Operation, consider creating a Map where the keys are codes and the values are Operations. This will make the conversion faster and the code cleaner.

Reply:
It will change start conditions